pkg/controller/parameters.go

```diff
@@ -132,3 +132,43 @@ func generateChecksumOfParameters(params map[string]interface{}) (string, error)
 	hash := sha256.Sum256(paramsAsJSON)
 	return fmt.Sprintf("%x", hash), nil
 }
+
+// prepareInProgressPropertyParameters generates the required parameters for setting
+// the in-progress status of a Type.
+// Returns (parameters, parametersChecksum, rawParametersWithRedaction, err)
```

Comment:
can you make the documentation clearer on what it returns clearer by explaining what each name is?